osu.Game/Overlays/Toolbar/ToolbarRulesetTabButton.cs

```diff
@@ -53,14 +55,32 @@ public RulesetButton()
                     Bottom = 5
                 };
             }
+            public string TooltipMainText { get; set; } = "";
 
             public bool Active
             {
                 set => Scheduler.AddOnce(() =>
                 {
                     if (value)
                     {
-                        IconContainer.Colour = Color4Extensions.FromHex("#00FFAA");
+                        string tooltipMainText = TooltipMainText;
+                        if (tooltipMainText == "osu!")
+                        {
+                            rulesetcolour = colours.PinkLight;
+                        }
+                        else if (tooltipMainText == "osu!mania")
+                        {
+                            rulesetcolour = colours.Purple1;
+                        }
+                        else if (tooltipMainText == "osu!catch")
+                        {
+                            rulesetcolour = colours.Blue;
+                        }
+                        else
+                        {
+                            rulesetcolour = colours.GreenLightest;
+                        }
+                        IconContainer.Colour = rulesetcolour;
```

Comment:
Regardless of whether this is going to be accepted as an idea (I think it's a bad idea and will be way too noisy visually, but will ultimately defer to @peppy / @arflyte / whoever), this cannot be written this way. Magic string comparisons such as this are a last resort.

*If* we decide to do this, this should probably be moved to an abstract property on `Ruleset`.

Reply:
Definitely.